server/src/main/java/org/opensearch/action/admin/cluster/stats/TransportClusterStatsAction.java

```diff
@@ -177,33 +202,41 @@ protected ClusterStatsNodeResponse nodeOperation(ClusterStatsNodeRequest nodeReq
             false
         );
         List<ShardStats> shardsStats = new ArrayList<>();
-        for (IndexService indexService : indicesService) {
-            for (IndexShard indexShard : indexService) {
-                if (indexShard.routingEntry() != null && indexShard.routingEntry().active()) {
-                    // only report on fully started shards
-                    CommitStats commitStats;
-                    SeqNoStats seqNoStats;
-                    RetentionLeaseStats retentionLeaseStats;
-                    try {
-                        commitStats = indexShard.commitStats();
-                        seqNoStats = indexShard.seqNoStats();
-                        retentionLeaseStats = indexShard.getRetentionLeaseStats();
-                    } catch (final AlreadyClosedException e) {
-                        // shard is closed - no stats is fine
-                        commitStats = null;
-                        seqNoStats = null;
-                        retentionLeaseStats = null;
+        if (!applyMetricFiltering || Metric.INDICES.containedIn(requestedMetrics)) {
+            CommonStatsFlags commonStatsFlags = new CommonStatsFlags();
+            for (String metric : nodeRequest.request.indicesMetrics()) {
+                if (INDEX_METRIC_TO_SHARDS_STATS_FLAG_MAP.containsKey(metric)) {
```

Comment:
Should we throw error if it does not contain ?

Reply:
All index metrics are not common stats, so can't throw an error here.